src/SourceBuild/content/repo-projects/aspnetcore.proj

```diff
@@ -58,4 +58,18 @@
     <ExtraPackageVersionPropsPackageInfo Include="MicrosoftCodeAnalysisVersion_LatestVS" Version="%24(MicrosoftCodeAnalysisCommonVersion)" />
   </ItemGroup>
 
+  <Target Name="SetNonShippingPackageSentinelVersions"
+    DependsOnTargets="GetProducedPackagesFromTransitiveReferences"
+    BeforeTargets="CreateBuildInputProps">
+    <ItemGroup>
+      <!--
+        aspnetcore uses the Microsoft.NETCore.BrowserDebugHost.Transport package version as a sentinel for the non-shipping runtime version.
+        As a result, we need to always provide it, even when we don't build that package (which is only built in browser-wasm-targeting builds).
+        However, aspnetcore doesn't have any other dependencies on a non-shipping package from dotnet/runtime.
+        -->
+      <_NonShippingPackageVersion Include="@(_DependencyProducedPackage->WithMetadataValue('OriginalItemSpec','runtime.proj')->WithMetadataValue('DotNetReleaseShipping','false')->Distinct())" />
+      <ExtraPackageVersionPropsPackageInfo Include="MicrosoftNETCoreBrowserDebugHostTransportVersion" Version="%(_NonShippingPackageVersion.Identity)" />
+    </ItemGroup>
+  </Target>
```

Comment:
This looks fine but I wonder if long term it would make sense for aspnetcore to depend on a transport package that represents their needs. What do they need the non-shipping version for?

Reply:
Aspnetcore is trying to download the archives from the blob storage (or the local storage in a VMR build). This storage is always versioned with the non-shipping version, even if the assets are shipping. Traditionally we've always used a non-shipping package that is used in the repository (in this case it's only used in some builds) to represent provide the non-shipping version.